File: backend/src/main/java/com/example/demo/Entity/Dipendente.java
package com.example.demo.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;


import java.util.List;

@Entity
@Getter
@Setter
public class Dipendente{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(unique = true)
    private String email;

    private long telefono;

    private String cognome;

    private String nome;

    private String sede;

    @OneToOne
    @JoinColumn
    private ContrattoLavorativo contrattoLavorativo;

    @ManyToOne
    @JoinColumn
    private Ruolo ruolo;

    @OneToMany(mappedBy = "dipendente")
    @JsonIgnore
    private List<R_TD> rtd;

    @OneToMany(mappedBy = "dipendente")
    @JsonIgnore
    private List<R_FD> rfd;

    //data assunzione

}

File: frontend/src/app/model/Dipendente.ts
import { ContrattoLavorativo } from "./ContrattoLavorativo";
import { GiornataFeriale } from "./GiornateFerie";
import { Ruolo } from "./Ruolo";

export interface Dipendente{
    id:number;
    nome:String;
    cognome:String;
    email:String;
    telefono:number;
    sede:String;
    contrattoLavorativo:ContrattoLavorativo;
    ruolo:Ruolo;
    giornateFeriali:GiornataFeriale[];
}

export class Dipendente implements Dipendente{

    id!:number;
    nome!:String;
    cognome!:String;
    email!:String;
    telefono!:number;
    sede!:String;
    contrattoLavorativo!:ContrattoLavorativo;
    ruolo!:Ruolo;
    giornateFeriali!:GiornataFeriale[];

    constructor(nome:String,cognome:String,ruolo:Ruolo,telefono:number,
         sede:String,email:String,contrattoLavorativo:ContrattoLavorativo){
        this.nome = nome;
        this.cognome = cognome; 
        this.ruolo = ruolo;
        this.telefono = telefono; 
        this.email = email;
        this.sede = sede;
        this.contrattoLavorativo = contrattoLavorativo;
    }
}